lib/src/main/java/io/ably/lib/realtime/Channel.java

```diff
@@ -475,6 +477,24 @@ private void failQueuedMessages(ErrorInfo reason) {
 		queuedMessages.clear();
 	}
 
+	private void replacePlaceholderParams(Param[] params) throws AblyException {
+		Param param;
+		for (int i = 0; i < params.length; i++) {
+			param = params[i];
+
+			if (KEY_UNTIL_ATTACH.equals(param.key) && "true".equals(param.value)) {
```

Comment:
I would argue that if you detect `untilAttach`, then it should be removed in all cases regardless of the value, and replaced with the serial when the value equals `true` (perhaps ignore case for convenience).  I don't see any value in sending the `untilAttach` param to the Ably service ever. Also, if we are raising an exception if `untilAttach` is called when the channel is attached, why don't we raise an exception if `untilAttach` is anything other than `true` / `false` (case-insensitive) with an error something along the lines `option untilAttach is invalid. "true" or "false" expected`